src/main/java/trackitnus/logic/parser/ParserUtil.java

```diff
@@ -143,6 +144,25 @@ public static Code parseCode(String code) throws ParseException {
         return new Code(trimmedCode);
     }
 
+    /**
+     * Parses an {@code Optional<String> code} into an {@code Code}.
+     *
+     * @throws ParseException if the given {@code code} is invalid.
+     */
+    public static Code parseOptionalCode(Optional<String> code) throws ParseException {
+        if (code.isEmpty()) {
+            return null;
+        }
+        String trimmedCode = code.get().trim();
+        if (!Address.isValidAddress(trimmedCode)) {
```

Comment:
This is indeed correct; however I believe it's better to have a separate function to handle this instead of importing from the irrelevant Address class.